src/crypto/OlmDevice.js

```diff
@@ -132,18 +143,51 @@ OlmDevice.prototype.init = async function() {
     this.deviceEd25519Key = e2eKeys.ed25519;
 };
 
-async function _initialiseAccount(cryptoStore, pickleKey, account) {
-    await cryptoStore.doTxn('readwrite', [IndexedDBCryptoStore.STORE_ACCOUNT], (txn) => {
-        cryptoStore.getAccount(txn, (pickledAccount) => {
-            if (pickledAccount !== null) {
-                account.unpickle(pickleKey, pickledAccount);
-            } else {
-                account.create();
-                pickledAccount = account.pickle(pickleKey);
-                cryptoStore.storeAccount(txn, pickledAccount);
-            }
+async function _initialiseAccount(fromExportedDevice, cryptoStore, pickleKey, account) {
```

Comment:
Given that the function is a big `if` statement conditioned on `fromExportedDevice`, it might make more sense to just leave `_initialiseAccount` as-is, and create a new function (maybe called `_initialiseAccountFormExportedDevice`) that loads the export?

Reply:
Done (used name `_initialiseFromExportedDevice` because we initialise more than just the account).